src/internal/operations/manifests_test.go

```diff
@@ -0,0 +1,612 @@
+			expectDCS: nil,
+		},
+		{
+			name: "don't get metadata, incomplete manifest",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mans:         []*kopia.ManifestEntry{makeMan(path.EmailCategory, "ir", "")},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   false,
+			assertErr: assert.NoError,
+			assertB:   assert.False,
+			expectDCS: nil,
+		},
+		{
+			name: "fetch manifests errors",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mrErr:        assert.AnError,
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   true,
+			assertErr: assert.Error,
+			assertB:   assert.False,
+			expectDCS: nil,
+		},
+		{
+			name: "verify distinct bases fails",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mans: []*kopia.ManifestEntry{
+					makeMan(path.EmailCategory, "", ""),
+					makeMan(path.EmailCategory, "", ""),
+				},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   true,
+			assertErr: assert.NoError, // No error, even though verify failed.
+			assertB:   assert.False,
+			expectDCS: nil,
+		},
+		{
+			name: "no manifests",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mans:         []*kopia.ManifestEntry{},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   true,
+			assertErr: assert.NoError,
+			assertB:   assert.True,
+			expectDCS: nil,
+		},
+		{
+			name: "only incomplete manifests",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{},
+				mans: []*kopia.ManifestEntry{
+					makeMan(path.EmailCategory, "ir", ""),
+					makeMan(path.ContactsCategory, "ir", ""),
+				},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   true,
+			assertErr: assert.NoError,
+			assertB:   assert.True,
+			expectDCS: nil,
+		},
+		{
+			name: "man missing backup id",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{colls: []data.Collection{}},
+				mans:         []*kopia.ManifestEntry{makeMan(path.EmailCategory, "", "")},
+			},
+			gdi:           mockGetDetailsIDer{detailsID: did},
+			reasons:       []kopia.Reason{},
+			getMeta:       true,
+			assertErr:     assert.Error,
+			assertB:       assert.False,
+			expectDCS:     []data.Collection{},
+			expectNilMans: true,
+		},
+		{
+			name: "one complete man, one incomplete",
+			mr: mockManifestRestorer{
+				mockRestorer: mockRestorer{colls: []data.Collection{}},
+				mans: []*kopia.ManifestEntry{
+					makeMan(path.EmailCategory, "", "bid"),
+					makeMan(path.EmailCategory, "ir", ""),
+				},
+			},
+			gdi:       mockGetDetailsIDer{detailsID: did},
+			reasons:   []kopia.Reason{},
+			getMeta:   true,
+			assertErr: assert.NoError,
+			assertB:   assert.True,
+			expectDCS: []data.Collection{},
```

Comment:
this should produce metadata no?

Reply:
They're not mocked in atm.  I'll make a second pass to handle that.  The table is a little wobbly since it assumes there's a single manifestRestorer, not one per manifest.  When I caught that I figured it'd be better to fix it in a follow up.